hedera-node/hedera-smart-contract-service-impl/src/test/java/com/hedera/node/app/service/contract/impl/test/exec/systemcontracts/hts/rejecttokens/RejectTokensDecoderTest.java

```diff
@@ -0,0 +1,253 @@
+import com.hedera.node.config.data.LedgerConfig;
+import com.swirlds.config.api.Configuration;
+import org.apache.tuweni.bytes.Bytes;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+public class RejectTokensDecoderTest {
+
+    @Mock
+    private HtsCallAttempt attempt;
+
+    @Mock
+    private AddressIdConverter addressIdConverter;
+
+    @Mock
+    private Enhancement enhancement;
+
+    @Mock
+    private HederaNativeOperations nativeOperations;
+
+    @Mock
+    private Configuration configuration;
+
+    @Mock
+    private LedgerConfig ledgerConfig;
+
+    private RejectTokensDecoder subject;
+
+    @BeforeEach
+    void setup() {
+        subject = new RejectTokensDecoder();
+
+        lenient().when(attempt.addressIdConverter()).thenReturn(addressIdConverter);
+    }
+
+    @Test
+    void decodeHtsCall() {
+        // given:
+        given(attempt.configuration()).willReturn(configuration);
+        given(configuration.getConfigData(LedgerConfig.class)).willReturn(ledgerConfig);
+        given(ledgerConfig.tokenRejectsMaxLen()).willReturn(10);
+        given(addressIdConverter.convert(asHeadlongAddress(SENDER_ID.accountNum())))
+                .willReturn(SENDER_ID);
+        final var encoded = Bytes.wrapByteBuffer(RejectTokensTranslator.TOKEN_REJECT.encodeCall(Tuple.of(
+                asHeadlongAddress(SENDER_ID.accountNum()),
+                new Address[] {FUNGIBLE_TOKEN_HEADLONG_ADDRESS},
+                new Tuple[] {})));
+
+        // when
+        given(attempt.inputBytes()).willReturn(encoded.toArrayUnsafe());
+
+        final var expected = TokenRejectTransactionBody.newBuilder()
+                .owner(SENDER_ID)
+                .rejections(TokenReference.newBuilder()
+                        .fungibleToken(FUNGIBLE_TOKEN_ID)
+                        .build())
+                .owner(SENDER_ID)
+                .build();
+
+        // then
+        final var decoded = subject.decodeTokenRejects(attempt);
+
+        assertNotNull(decoded);
+        assertEquals(expected, decoded.tokenReject());
+    }
+
+    @Test
+    void decodeHtsCallNFT() {
+        // given:
+        given(attempt.configuration()).willReturn(configuration);
+        given(configuration.getConfigData(LedgerConfig.class)).willReturn(ledgerConfig);
+        given(ledgerConfig.tokenRejectsMaxLen()).willReturn(10);
+        given(addressIdConverter.convert(asHeadlongAddress(SENDER_ID.accountNum())))
+                .willReturn(SENDER_ID);
+
+        final var encoded = Bytes.wrapByteBuffer(RejectTokensTranslator.TOKEN_REJECT.encodeCall(
+                Tuple.of(asHeadlongAddress(SENDER_ID.accountNum()), new Address[] {}, new Tuple[] {
+                    Tuple.of(FUNGIBLE_TOKEN_HEADLONG_ADDRESS, 1L)
```

Comment:
Looks funny in this test to use `FUNGIBLE_TOKEN_{ID,HEADLONG_ADDRESS}` - isn't there a different constant you can use?  (Obviously doesn't matter when you _execute_ this unit test, but it's odd to see.)

Reply:
Yep, changed to NFT address